crates/lib/src/bootc_composefs/gc.rs

```diff
@@ -0,0 +1,226 @@
+//! This module handles the case when deleting a deployment fails midway
+//!
+//! There could be the following cases (See ./delete.rs:delete_composefs_deployment):
+//! - We delete the bootloader entry but fail to delete image
+//! - We delete bootloader + image but fail to delete the state/unrefenced objects etc
+
+use anyhow::{Context, Result};
+use cap_std_ext::{
+    cap_std::{ambient_authority, fs::Dir},
+    dirext::CapStdExtDirExt,
+};
+use composefs::fsverity::{FsVerityHashValue, Sha512HashValue};
+
+use crate::{
+    bootc_composefs::{
+        boot::{get_esp_partition, get_sysroot_parent_dev, mount_esp},
+        delete::{delete_image, delete_staged, delete_state_dir, get_image_objects},
+        status::{
+            composefs_deployment_status, get_bootloader, get_sorted_grub_uki_boot_entries,
+            get_sorted_type1_boot_entries,
+        },
+    },
+    composefs_consts::{STATE_DIR_RELATIVE, USER_CFG},
+    spec::Bootloader,
+};
+
+#[fn_error_context::context("Listing EROFS images")]
+fn list_erofs_images(sysroot: &Dir) -> Result<Vec<String>> {
+    let images_dir = sysroot
+        .open_dir("composefs/images")
```

Comment:
Hmm there isn't an API for this in `Repository`?

I think what would help here is if this API actually took a `Repository` instance for now at least - we can reference its fd and read things directly. Same elsewehere